### PR TITLE
docs: add initial guide about debugging monitor mode

### PR DESCRIPTION
**Pull request by jedevc**
Adds some very basic early documentation about monitor mode and debugging - the idea of this is to link to it in our release notes for buildx v0.11, so that we can start getting early feedback from users and work out what we need to focus on in the future :tada:

Not sure where this doc should go actually... we've moved the guides into the docs repo - but ideally we should keep it in buildx, at least while we're quickly iterating on this feature.

### CODE REVIEW

**Comment by crazy-max** [comment]
> Not sure where this doc should go actually... we've moved the guides into the docs repo - but ideally we should keep it in buildx, at least while we're quickly iterating on this feature.

That's fine during the experimental phase but should be in docs repo on GA.

File: docs/guides/debugging.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+/ # ls
+bin    etc    lib    mnt    proc   run    srv    tmp    var
+dev    home   media  opt    root   sbin   sys    usr    work
+```
+
+## Using the monitor
+
+By default, when debugging, you'll be dropped into a shell in the final stage.
+
+By using the `Ctrl-a-c` key combination when in a debugging shell, you'll
```
Comment (dvdksn) [inline docs/guides/debugging.md:56]: I'm guessing `a` is for `Alt`?

```suggestion
When you're in a debug shell, you can use the `Ctrl-Alt-C` key combination to
```

File: docs/guides/debugging.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+toggle between this debug shell and the monitor mode for the debugging session,
+which allows you to run high-level commands to control the debug environment.
```
Comment (dvdksn) [inline docs/guides/debugging.md:58]: I'm guessing `a` is for `Alt`?

```suggestion
When you're in a debug shell, you can use the `Ctrl-Alt-C` key combination to
toggle between the debug shell and the monitor mode.
In monitor mode, you can run commands that control the debug environment.
```

File: docs/guides/debugging.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+For example, you can use `ls` to see the contents of the image:
+
+```console
+/ # ls
+bin    etc    lib    mnt    proc   run    srv    tmp    var
+dev    home   media  opt    root   sbin   sys    usr    work
+```
+
+## Using the monitor
```
Comment (dvdksn) [inline docs/guides/debugging.md:52]:
```suggestion
## Monitor mode
```

File: docs/guides/debugging.md (hunk 2 above)
Comment (jedevc) [reply docs/guides/debugging.md:58]: Ah it's actually not - it's CTRL+A, and then C on the keyboard. https://github.com/docker/buildx/pull/1168#pullrequestreview-1006893745.

I *think* this is inspired by QEMU. CTRL+A enters a "command" view, and then next character indicates what to do, e.g. `c` might enter a console, `x` might kill the process, etc.

Good callout, I'm not sure how best to phrase this clearly :cry:

Comment (dvdksn) [reply docs/guides/debugging.md]: OK, got it. Sorry for the confusion. Not sure what the best way to put this is either. Vimmer in me wants to put something like:

the key combination `<C-a>c` (`Ctrl+A` followed by a `C`) opens the monitor mode

Comment (jedevc) [reply docs/guides/debugging.md]: > When you're in a debug shell, you can use the `Ctrl-a-c` key combination (press
`Ctrl`+`a` together, lift, then press `c`) to toggle between the debug shell
and the monitor mode. In monitor mode, you can run commands that control the
debug environment.

Hopefully that's descriptive enough :tada:

**Review by dvdksn — APPROVED** [review]
LGTM

**Review by crazy-max — APPROVED** [review]

**Pull request merged**